pkg/audit/verifier.go

```diff
@@ -177,6 +177,11 @@ func (verifier *Verifier) Verify(ctx context.Context, stripe *Stripe, skip map[s
 	for _, pieceNum := range pieceNums {
 		failedNodes = append(failedNodes, shares[pieceNum].NodeID)
 	}
+	//remove failed audit pieces from the pointer so as to only penalize once for failed audits
+	err = verifier.RemoveFailedPieces(ctx, stripe.SegmentPath, stripe.Segment, failedNodes)
+	if err != nil {
+		verifier.log.Warn("Verify: failed to delete failed pieces", zap.Error(err))
```

Comment:
Instead of logging the error, let's return it on line 239, and combine it with the error returned on line 231.

Reply:
I'm not entirely sure I agree

Reply:
Why don't you? Note that in this method it is perfectly OK to return both an error and a report. The caller will process the report despite the error.

Reply:
What would we do differently if we know that we failed to update the pointer?  Stop the service?  Simply log it somewhere else?  I don't understand the value.

Reply:
OK, you have a point. Let's leave it like this.